src/navigation/machines/oid4vciStateNavigation.tsx

```diff
@@ -269,6 +275,44 @@ const navigateReviewCredentials = async (args: OID4VCIMachineNavigationArgs): Pr
 
   const onDecline = async (): Promise<void> => {
     oid4vciMachine.send(OID4VCIMachineEvents.DECLINE);
+
+    // FIXME temp solution to have activity for oid4vci-holder, we should add this to the plugin later
+    function determineCredentialDocumentFormat(documentFormat: DocumentFormat): CredentialDocumentFormat {
+      switch (documentFormat) {
+        case DocumentFormat.JSONLD:
+          return CredentialDocumentFormat.JSON_LD;
+        case DocumentFormat.JWT:
+          return CredentialDocumentFormat.JWT;
+        case DocumentFormat.SD_JWT_VC:
+          return CredentialDocumentFormat.SD_JWT;
+        case DocumentFormat.MSO_MDOC:
+          return CredentialDocumentFormat.MSO_MDOC;
+        default:
+          throw new Error(`Not supported document format: ${documentFormat}`);
+      }
+    }
+
+    // FIXME temp solution to have activity for oid4vci-holder, we should add this to the plugin later
+    store.dispatch<any>(
+      storeActivityLogging({
+        level: LogLevel.INFO,
+        system: System.OID4VCI,
+        subSystemType: SubSystem.VC_ISSUER,
+        initiatorType: InitiatorType.SYSTEM,
+        description: 'decline credential',
+        actionType: ActionType.READ,
+        actionSubType: DefaultActionSubType.VC_ISSUE_DECLINE,
+        // @ts-ignore
+        credentialType: determineCredentialDocumentFormat(CredentialMapper.detectDocumentType(credentialsToAccept[0].rawVerifiableCredential)),
+        // @ts-ignore
+        credentialHash: credentialsToAccept[0].uniformVerifiableCredential.id ?? computeEntryHash(credentialsToAccept[0].uniformVerifiableCredential),
+        originalCredential: JSON.stringify(credentialsToAccept[0].uniformVerifiableCredential),
+        // @ts-ignore
+        partyCorrelationType: contact?.identities[0].identifier.type, // TODO fix types
```

Comment:
Are we 100% sure there will be identities? contact?.identities[0]?.identifier.type or similar contact?

Reply:
yes, because every contact has at least 1 or more identities